File: server/php/src/Application/DTOs/PaymentLink/RequestDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\PaymentLink;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;
use OnlinePayments\ExampleApp\Application\Domain\Enums\PaymentLinks\Action;
use OnlinePayments\ExampleApp\Application\Domain\Enums\PaymentLinks\ValidFor;

class RequestDto
{
    public ?int $amount;
    public ?Currency $currency;
    public ?string $description = null;
    public ?ValidFor $validFor;
    public ?Action $action;

    public function __construct(
        ?int $amount,
        ?Currency $currency,
        ?string $description = null,
        ?ValidFor $validFor = null,
        ?Action $action = null
    ) {
        $this->amount = $amount;
        $this->currency = $currency;
        $this->description = $description;
        $this->validFor = $validFor;
        $this->action = $action;
    }
}

File: server/ruby/app/business/dtos/payment_link/request_dto.rb
module Business
  module Dtos
    module PaymentLink
      class RequestDto
        attr_accessor :amount, :currency, :description, :valid_for, :action

        def initialize( amount: nil, currency: nil, description: nil, valid_for: nil, action: nil)
          @amount       = amount
          @currency     = currency
          @description  = description
          @valid_for    = valid_for
          @action       = action
        end
      end
    end
  end
end